### PR TITLE
Fix prune slab edge case

### PR DESCRIPTION
**Pull request by ChrisSchinnerl**
f/u to 610 which adds a regression test

Also fixes a bug related to the official amazon client and multipart uploads. Reason for that bug was that different clients seem to behave differently around passing etags to `CompleteMultipartUpload`. So sometimes the etag would reach the backend with quotes surrounding it and sometimes not.

### CODE REVIEW

File: stores/multipart.go [hunk 1]
```diff
@@ -274,7 +275,7 @@ func (s *SQLStore) CompleteMultipartUpload(ctx context.Context, bucket, path str
 				} else if mu.Parts[j].PartNumber > part.PartNumber {
 					// missing part
 					return api.ErrPartNotFound
-				} else if mu.Parts[j].PartNumber == part.PartNumber && mu.Parts[j].Etag == part.ETag {
+				} else if mu.Parts[j].PartNumber == part.PartNumber && mu.Parts[j].Etag == strings.Trim(part.ETag, "\"") {
```
Comment (peterjan) [inline stores/multipart.go:278]: I think this could've been stripped properly in `backend.go`? 
Should we add some type of etag sanitizer?

File: worker/client.go [hunk 2]
```diff
@@ -243,7 +243,7 @@ func (c *Client) UploadMultipartUploadPart(ctx context.Context, r io.Reader, pat
 		err, _ := io.ReadAll(resp.Body)
 		return "", errors.New(string(err))
 	}
-	return resp.Header.Get("ETag"), nil
+	return strings.Trim(resp.Header.Get("ETag"), "\""), nil
```
Comment (peterjan) [inline worker/client.go:246]: Oh you are probably doing it here right? If that's the case that's ok.

**Review by peterjan — APPROVED** [review]
